diff --git a/targets/GD32F303RGT6_BearPi/liteos.ld b/targets/GD32F303RGT6_BearPi/liteos.ld
index 4f8c26b2..b9d765c4 100644
--- a/targets/GD32F303RGT6_BearPi/liteos.ld
+++ b/targets/GD32F303RGT6_BearPi/liteos.ld
@@ -1,5 +1,6 @@
 ENTRY(Reset_Handler)
 
+_sstack = 0x20000000;
 _estack = 0x20018000;
 __los_stack_size__ = 0x400;
 
@@ -46,6 +47,34 @@ SECTIONS
     __rodata_end = .;
 } >FLASH
 
+.preinit_array     :
+{
+  . = ALIGN(4);
+  PROVIDE_HIDDEN (__preinit_array_start = .);
+  KEEP (*(.preinit_array*))
+  PROVIDE_HIDDEN (__preinit_array_end = .);
+  . = ALIGN(4);
+} >FLASH
+
+.init_array :
+{
+  . = ALIGN(4);
+  PROVIDE_HIDDEN (__init_array_start = .);
+  KEEP (*(SORT(.init_array.*)))
+  KEEP (*(.init_array*))
+  PROVIDE_HIDDEN (__init_array_end = .);
+  . = ALIGN(4);
+} >FLASH
+.fini_array :
+{
+  . = ALIGN(4);
+  PROVIDE_HIDDEN (__fini_array_start = .);
+  KEEP (*(SORT(.fini_array.*)))
+  KEEP (*(.fini_array*))
+  PROVIDE_HIDDEN (__fini_array_end = .);
+  . = ALIGN(4);
+} >FLASH
+
 _si_liteos_vector_data = LOADADDR(.vector_ram);
 
 .vector_ram :
